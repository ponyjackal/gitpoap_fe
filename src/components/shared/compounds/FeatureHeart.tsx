import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Loader } from '@mantine/core';
import { FaHeart } from 'react-icons/fa';
import { ExtraHover, ExtraPressed, TextAccent, TextLight } from '../../../colors';
import { useFeaturedPOAPs, useFeaturedPOAPsDispatch } from '../../profile/FeaturedPOAPsContext';

type Props = {
  className?: string;
  poapTokenId: string;
};

const HeartWrapper = styled.div<{ isFeatured: boolean }>`
  > svg {
    fill: ${({ isFeatured }) => (isFeatured ? TextAccent : TextLight)};
  }
`;

const FeatureHeartStyled = styled(FaHeart)`
  color: ${TextLight};
  height: ${rem(18)};
  width: ${rem(18)};
  transition: 150ms fill ease;
  fill: ${TextLight};

  &:hover {
    fill: ${ExtraHover};
    cursor: pointer;
  }
  &:active {
    fill: ${ExtraPressed};
  }
`;

export const FeatureHeart = ({ className, poapTokenId }: Props) => {
  const {
    featuredPOAPsState: { featuredPOAPTokenIDs },
    showHearts,
    loadingIds,
  } = useFeaturedPOAPs();
  const { addFeaturedPOAP, removeFeaturedPOAP } = useFeaturedPOAPsDispatch();
  const isFeatured: boolean = !!featuredPOAPTokenIDs && !!featuredPOAPTokenIDs[poapTokenId];
  const isFeaturedLoading = !!loadingIds && !!loadingIds[poapTokenId];

  if (!showHearts) {
    return null;
  }

  return (
    <HeartWrapper isFeatured={isFeatured}>
      {isFeaturedLoading ? (
        <Loader className={className} size={18} />
      ) : (
        <FeatureHeartStyled
          className={className}
          onClick={() => {
            if (isFeatured) {
              removeFeaturedPOAP(poapTokenId);
            } else {
              addFeaturedPOAP(poapTokenId);
            }
          }}
        />
      )}
    </HeartWrapper>
  );
};
