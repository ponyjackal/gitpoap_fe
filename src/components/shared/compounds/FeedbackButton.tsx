import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Popover, Box } from '@mantine/core';
import { useDisclosure } from '@mantine/hooks';
import { GitPOAPLogoNoText } from '../elements/icons';
import { BackgroundPanel, BackgroundPanel2, ExtraHover, ExtraPressed } from '../../../colors';
import { Text } from '../elements';

const StyledLogo = styled(GitPOAPLogoNoText)`
  height: ${rem(45)};
  width: ${rem(45)};
  transition: transform 200ms ease-in-out;
  path {
    transition: fill 200ms ease-in-out;
  }
`;

const CircleButton = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${rem(70)};
  width: ${rem(70)};
  background-color: ${BackgroundPanel};
  transition: background-color 200ms ease-in-out, transform 200ms ease-in-out;
  border-radius: 50%;
  border: ${rem(3)} solid ${BackgroundPanel2};
  &:hover {
    background-color: ${BackgroundPanel2};
    transform: scale(1.1);
    cursor: pointer;
    ${StyledLogo} {
      transform: scale(1.1);
      path {
        fill: ${ExtraHover};
      }
    }
  }
  &:active {
    background-color: ${BackgroundPanel};
    ${StyledLogo} {
      color: ${ExtraHover};
      path {
        fill: ${ExtraPressed};
      }
    }
  }
`;

const PopoverContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

type Props = {
  className?: string;
  href: string;
};

export const FeedbackButton = ({ className, href }: Props) => {
  const [isOpen, { open, close }] = useDisclosure(false);
  return (
    <Popover
      opened={isOpen}
      onClose={close}
      position="top"
      withArrow
      trapFocus={false}
      closeOnEscape={false}
      transition="fade"
      transitionDuration={200}
      radius="lg"
      withinPortal
      zIndex={100}
      styles={{
        dropdown: {
          backgroundColor: BackgroundPanel,
        },
      }}
    >
      <Popover.Target>
        <Box onMouseEnter={open} onMouseLeave={close} className={className} style={{ zIndex: 100 }}>
          <CircleButton href={href} target="_blank" rel="noopener noreferrer">
            <StyledLogo />
          </CircleButton>
        </Box>
      </Popover.Target>
      <Popover.Dropdown>
        <PopoverContainer>
          <Text>{'Give Feedback'}</Text>
        </PopoverContainer>
      </Popover.Dropdown>
    </Popover>
  );
};
