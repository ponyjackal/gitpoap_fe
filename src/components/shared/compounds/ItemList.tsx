import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Header } from '../elements/Header';
import { Button } from '../elements/Button';
import { FaPlus } from 'react-icons/fa';
import { Select } from '../elements/Select';
import { Text } from '../elements/Text';
import { TextGray } from '../../../colors';
import { Input } from '../elements/Input';

type Props = {
  title?: string;
  className?: string;
  children?: React.ReactNode;
  selectOptions: SelectOption[];
  selectValue: string;
  onSelectChange: (value: string) => void;
  isLoading: boolean;
  hasShowMoreButton: boolean;
  showMoreOnClick: () => void;
  searchInputPlaceholder?: string;
  searchInputValue?: string;
  onSearchInputChange?: React.ChangeEventHandler<HTMLInputElement>;
};

export type SelectOption<T = string> = { value: T; label: string };

const Container = styled.div`
  display: inline-flex;
  flex-direction: column;
  width: 100%;
`;

const Heading = styled.div`
  display: inline-flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const ListTitle = styled(Header)`
  font-size: ${rem(30)};
  line-height: ${rem(42)};
`;

const Sorting = styled.div`
  display: inline-flex;
  flex-direction: row;
`;

const ShowMore = styled(Button)`
  align-self: center;
`;

const SortBy = styled(Text)`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  text-transform: uppercase;
  color: ${TextGray};
  margin-right: ${rem(10)};
`;

const SearchInput = styled(Input)`
  margin-right: ${rem(50)};
`;

export const ItemList = ({
  children,
  className,
  title,
  selectOptions,
  selectValue,
  onSelectChange,
  isLoading,
  hasShowMoreButton,
  showMoreOnClick,
  searchInputPlaceholder,
  searchInputValue,
  onSearchInputChange,
}: Props) => {
  return (
    <Container className={className}>
      <Heading>
        <ListTitle>{title}</ListTitle>
        <Sorting>
          {searchInputValue !== undefined && onSearchInputChange && (
            <SearchInput
              placeholder={searchInputPlaceholder}
              value={searchInputValue}
              onChange={onSearchInputChange}
            />
          )}
          <SortBy>{'Sort By: '}</SortBy>
          <Select data={selectOptions} value={selectValue} onChange={onSelectChange} />
        </Sorting>
      </Heading>
      {children}
      {hasShowMoreButton && (
        <ShowMore
          onClick={showMoreOnClick}
          leftIcon={<FaPlus />}
          variant="outline"
          loading={isLoading}
        >
          {'Show more'}
        </ShowMore>
      )}
    </Container>
  );
};
