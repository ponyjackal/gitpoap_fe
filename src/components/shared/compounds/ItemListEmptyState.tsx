import React from 'react';
import { rem } from 'polished';
import { Group } from '@mantine/core';

type Props = {
  icon?: React.ReactNode;
  children: React.ReactNode;
};

export const EmptyState = ({ children, icon }: Props) => {
  return (
    <Group
      direction="column"
      position="center"
      align="center"
      grow
      style={{ padding: rem(60), flex: '1', gridColumnEnd: 1, gridRowEnd: 2 }}
    >
      {icon && icon}
      {children}
    </Group>
  );
};
