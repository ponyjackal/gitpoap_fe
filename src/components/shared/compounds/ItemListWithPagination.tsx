import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Box, BoxProps, Pagination, Group, Loader } from '@mantine/core';
import { useMediaQuery } from '@mantine/hooks';
import { Header, Select, Text, Input } from '../elements';
import { TextGray } from '../../../colors';
import { BREAKPOINTS } from '../../../constants';

type Props = BoxProps & {
  title?: string;
  className?: string;
  children?: React.ReactNode;
  selectOptions: SelectOption[];
  selectValue: string;
  onSelectChange: (value: string) => void;
  isLoading: boolean;
  page: number;
  totalPage: number;
  handlePageChange: (page: number) => void;
  searchInputPlaceholder?: string;
  searchInputValue?: string;
  onSearchInputChange?: React.ChangeEventHandler<HTMLInputElement>;
};

export type SelectOption<T = string> = { value: T; label: string };

const Container = styled(Box)<BoxProps>`
  display: inline-flex;
  flex-direction: column;
  width: 100%;
`;

const Heading = styled.div`
  display: inline-flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: ${rem(30)};
`;

const ListTitle = styled(Header)`
  font-size: ${rem(30)};
  line-height: ${rem(42)};

  @media (max-width: ${BREAKPOINTS.sm}px) {
    font-size: ${rem(26)};
    line-height: ${rem(32)};
  }
`;

const Sorting = styled.div`
  display: inline-flex;
  flex-direction: row;
`;

const SortBy = styled(Text)`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  text-transform: uppercase;
  color: ${TextGray};
  margin-right: ${rem(10)};
`;

const SearchInput = styled(Input)`
  margin-right: ${rem(50)};
`;

export const ItemListWithPagination = ({
  children,
  className,
  title,
  selectOptions,
  selectValue,
  onSelectChange,
  isLoading,
  page,
  totalPage,
  handlePageChange,
  searchInputPlaceholder,
  searchInputValue,
  onSearchInputChange,
  ...boxProps
}: Props) => {
  const matchesBreakpointSmall = useMediaQuery(`(max-width: ${rem(BREAKPOINTS.sm)})`, false);

  return (
    <Container className={className} {...boxProps}>
      <Heading>
        <ListTitle>{title}</ListTitle>
        <Sorting>
          {!matchesBreakpointSmall && searchInputValue !== undefined && onSearchInputChange && (
            <SearchInput
              placeholder={searchInputPlaceholder}
              value={searchInputValue}
              onChange={onSearchInputChange}
            />
          )}
          {!matchesBreakpointSmall && <SortBy>{'Sort By: '}</SortBy>}
          <Select data={selectOptions} value={selectValue} onChange={onSelectChange} />
        </Sorting>
      </Heading>
      {isLoading ? (
        <Group position="center" align="center" grow>
          <Loader size="xl" variant="dots" />
        </Group>
      ) : (
        children
      )}
      <Pagination
        position="center"
        page={page}
        onChange={handlePageChange}
        total={totalPage}
        mt={rem(20)}
      />
    </Container>
  );
};
