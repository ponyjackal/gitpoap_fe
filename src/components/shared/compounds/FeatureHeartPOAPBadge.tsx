import React from 'react';
import { Loader } from '../elements/Loader';
import { useFeaturedPOAPsDispatch } from '../../profile/FeaturedPOAPsContext';
import { HeartWrapper, FeatureHeartStyled } from './FeatureHeart';

type Props = {
  className?: string;
  poapTokenId: string;
  isFeatured: boolean;
  isFeaturedLoading: boolean;
};

export const FeaturedHeartPOAPBadge = ({
  className,
  poapTokenId,
  isFeatured,
  isFeaturedLoading,
}: Props) => {
  const { addFeaturedPOAP, removeFeaturedPOAP } = useFeaturedPOAPsDispatch();
  return (
    <HeartWrapper isFeatured={isFeatured}>
      {isFeaturedLoading ? (
        <Loader className={className} size={18} />
      ) : (
        <FeatureHeartStyled
          className={className}
          onClick={() => {
            if (isFeatured) {
              removeFeaturedPOAP(poapTokenId);
            } else {
              addFeaturedPOAP(poapTokenId);
            }
          }}
        />
      )}
    </HeartWrapper>
  );
};
