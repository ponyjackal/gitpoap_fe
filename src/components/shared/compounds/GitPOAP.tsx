import { Link } from '../../Link';
import React from 'react';
import styled, { css } from 'styled-components';
import { rem } from 'polished';
import { GitPOAPBadge } from '../elements/GitPOAPBadge';
import { Title } from '../elements/Title';
import { TextLight } from '../../../colors';
import { FeatureHeart } from './FeatureHeart';

type Props = {
  gitPOAPId: number;
  imgSrc: string;
  name: string;
  orgName: string;
  description?: string;
  className?: string;
  poapTokenId?: string;
  onClick?: () => void;
};

const LineClamp = (lines: number) => css`
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: ${lines};
  -webkit-box-orient: vertical;
`;

const Wrapper = styled.div`
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const Info = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: ${rem(150)};
`;

const TitleStyled = styled(Title)`
  margin-top: ${rem(10)};
  text-align: center;
  ${LineClamp(3)};
`;

const OrgName = styled.div`
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(11)};
  line-height: ${rem(18)};
  text-align: center;
  letter-spacing: ${rem(1.2)};
  text-transform: uppercase;
  color: ${TextLight};
  margin-top: ${rem(8)};
`;

const Description = styled.div`
  font-family: PT Mono;
  font-style: normal;
  font-weight: normal;
  font-size: ${rem(11)};
  line-height: ${rem(14)};
  text-align: center;
  letter-spacing: ${rem(-0.1)};
  color: ${TextLight};
  margin-top: ${rem(8)};
  ${LineClamp(2)};
`;

const Heart = styled(FeatureHeart)`
  position: absolute;
  bottom: ${rem(0)};
  right: ${rem(10)};
`;

const BadgeWrapper = styled(Wrapper)`
  position: relative;
`;

export const GitPOAP = ({
  className,
  poapTokenId,
  gitPOAPId,
  imgSrc,
  name,
  orgName,
  description,
  onClick,
}: Props) => {
  return (
    <Wrapper className={className}>
      <BadgeWrapper>
        <Link href={`/gp/${gitPOAPId}`} passHref>
          <GitPOAPBadge
            size="sm"
            imgUrl={imgSrc}
            onClick={() => {
              if (onClick) {
                onClick();
              }
            }}
          />
          {poapTokenId && <Heart poapTokenId={poapTokenId} />}
        </Link>
      </BadgeWrapper>
      <Info>
        <Link href={`/gp/${gitPOAPId}`} passHref>
          <TitleStyled>{name}</TitleStyled>
        </Link>
        <OrgName>{orgName}</OrgName>
        {description && <Description>{description}</Description>}
      </Info>
    </Wrapper>
  );
};
