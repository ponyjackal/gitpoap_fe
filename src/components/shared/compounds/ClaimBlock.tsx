import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { Button } from '../elements/Button';
import { GitPOAP } from './GitPOAP';
import { FaCheckCircle, FaCoins, FaEthereum } from 'react-icons/fa';

type Props = {
  gitPOAPId: number;
  imgSrc: string;
  name: string;
  orgName: string;
  description: string;
  onClickClaim: () => void;
  onClickBadge?: () => void;
  isClaimed?: boolean;
  isLoading?: boolean;
  isConnected?: boolean;
};

const Wrapper = styled.div`
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  height: ${rem(340)};
`;

const ButtonWrapper = styled.div`
  flex: 1;
  margin-top: ${rem(14)};
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
`;

const getButtonText = (isClaimed: boolean | undefined, isConnected: boolean | undefined) => {
  if (isClaimed) {
    return 'Minted';
  }

  if (!isConnected) {
    return 'Connect';
  }

  return 'Mint';
};

export const ClaimBlock = ({
  gitPOAPId,
  imgSrc,
  name,
  orgName,
  description,
  onClickClaim,
  onClickBadge,
  isClaimed,
  isLoading,
  isConnected,
}: Props) => {
  return (
    <Wrapper>
      <GitPOAP
        gitPOAPId={gitPOAPId}
        imgSrc={imgSrc}
        name={name}
        orgName={orgName}
        description={description}
        onClick={onClickBadge}
      />
      <ButtonWrapper>
        <Button
          onClick={onClickClaim}
          loading={isLoading}
          leftIcon={isClaimed ? <FaCheckCircle /> : !isConnected ? <FaEthereum /> : <FaCoins />}
          disabled={isClaimed}
        >
          {getButtonText(isClaimed, isConnected)}
        </Button>
      </ButtonWrapper>
    </Wrapper>
  );
};
