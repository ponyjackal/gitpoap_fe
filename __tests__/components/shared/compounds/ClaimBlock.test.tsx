import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { ClaimBlock } from '../../../../src/components/shared/compounds/ClaimBlock';

describe('ClaimBlock', () => {
  it('should render a button with "Minted" text', () => {
    render(
      <ClaimBlock
        gitPOAPId={1}
        imgSrc="https://poap.gallery/static/media/test_gitpoap.1b1b1b1b.png"
        name={'GitPOAP={2017 OpenZeppelin Contracts Contributor'}
        orgName={'Polygon'}
        description={'To the creators of Polygon Network'}
        isClaimingAll={false}
        onClickClaim={() => {}}
        isClaimed={true}
      />,
    );
    const button = screen.getByRole('button', { name: 'Minted' });

    expect(button).toHaveTextContent('Minted');
    expect(button).toBeDisabled();
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });

  it('should render a button with "Mint" text', () => {
    render(
      <ClaimBlock
        gitPOAPId={1}
        imgSrc="https://poap.gallery/static/media/test_gitpoap.1b1b1b1b.png"
        name={'GitPOAP={2017 OpenZeppelin Contracts Contributor'}
        orgName={'Polygon'}
        description={'To the creators of Polygon Network'}
        isClaimingAll={true}
        onClickClaim={() => {}}
        isClaimed={false}
      />,
    );
    const button = screen.getByRole('button', { name: 'Mint' });

    expect(button).toHaveTextContent('Mint');
    expect(button).toBeDisabled();
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });
});
