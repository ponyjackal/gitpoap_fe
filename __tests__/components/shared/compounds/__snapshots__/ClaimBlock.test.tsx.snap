// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimBlock should render a button with "Mint" text 1`] = `
.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 16px;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 36px;
  padding-left: 12px;
  padding-right: 18px;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  display: inline-block;
  width: auto;
  border-radius: 4px;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 1px solid transparent;
  background-color: #228be6;
  color: #fff;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-2:hover {
    background-color: #1c7ed6;
  }
}

@media (hover: none) {
  .emotion-2:active {
    background-color: #1c7ed6;
  }
}

.emotion-2:active {
  -webkit-transform: translateY(1px);
  -moz-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.emotion-2:disabled,
.emotion-2[data-disabled] {
  border-color: transparent;
  background-color: #e9ecef;
  color: #adb5bd;
  cursor: not-allowed;
  background-image: none;
  pointer-events: none;
}

.emotion-2:disabled:active,
.emotion-2[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2[data-loading] {
  pointer-events: none;
}

.emotion-2[data-loading]::before {
  content: "";
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 4px;
  cursor: not-allowed;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 10px;
}

.emotion-8 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  data-button="true"
  data-disabled="true"
  disabled=""
  id="rewardId-1"
  type="button"
>
  <div
    class="emotion-3 emotion-4"
  >
    <span
      class="emotion-5 emotion-6 emotion-7"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 405.3V448c0 35.3 86 64 192 64s192-28.7 192-64v-42.7C342.7 434.4 267.2 448 192 448S41.3 434.4 0 405.3zM320 128c106 0 192-28.7 192-64S426 0 320 0 128 28.7 128 64s86 64 192 64zM0 300.4V352c0 35.3 86 64 192 64s192-28.7 192-64v-51.6c-41.3 34-116.9 51.6-192 51.6S41.3 334.4 0 300.4zm416 11c57.3-11.1 96-31.7 96-55.4v-42.7c-23.2 16.4-57.3 27.6-96 34.5v63.6zM192 160C86 160 0 195.8 0 240s86 80 192 80 192-35.8 192-80-86-80-192-80zm219.3 56.3c60-10.8 100.7-32 100.7-56.3v-42.7c-35.5 25.1-96.5 38.6-160.7 41.8 29.5 14.3 51.2 33.5 60 57.2z"
        />
      </svg>
    </span>
    <span
      class="emotion-8 emotion-9"
    >
      Mint
    </span>
  </div>
</button>
`;

exports[`ClaimBlock should render a button with "Minted" text 1`] = `
.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 16px;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 36px;
  padding-left: 12px;
  padding-right: 18px;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  display: inline-block;
  width: auto;
  border-radius: 4px;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 1px solid transparent;
  background-color: #228be6;
  color: #fff;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-2:hover {
    background-color: #1c7ed6;
  }
}

@media (hover: none) {
  .emotion-2:active {
    background-color: #1c7ed6;
  }
}

.emotion-2:active {
  -webkit-transform: translateY(1px);
  -moz-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.emotion-2:disabled,
.emotion-2[data-disabled] {
  border-color: transparent;
  background-color: #e9ecef;
  color: #adb5bd;
  cursor: not-allowed;
  background-image: none;
  pointer-events: none;
}

.emotion-2:disabled:active,
.emotion-2[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2[data-loading] {
  pointer-events: none;
}

.emotion-2[data-loading]::before {
  content: "";
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 4px;
  cursor: not-allowed;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 10px;
}

.emotion-8 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  data-button="true"
  data-disabled="true"
  disabled=""
  id="rewardId-1"
  type="button"
>
  <div
    class="emotion-3 emotion-4"
  >
    <span
      class="emotion-5 emotion-6 emotion-7"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        />
      </svg>
    </span>
    <span
      class="emotion-8 emotion-9"
    >
      Minted
    </span>
  </div>
</button>
`;
